lib/Unexpected.js

```diff
@@ -484,14 +482,30 @@ Unexpected.prototype.fail = function (arg) {
                 output.error('Explicit failure');
             }
         };
+        var expect = this.expect;
         Object.keys(arg).forEach(function (key) {
             var value = arg[key];
             if (key === 'diff') {
-                error.createDiff = value;
+                if (typeof value === 'function' && this.parent) {
+                    error.createDiff = function (output, diff, inspect, equal) {
+                        var inheritedOutput = expect.createOutput(output.format);
+                        inheritedOutput.inline = output.inline;
+                        inheritedOutput.output = output.output;
```

Comment:
#273 and #332 were steps in this direction. I think it's time (11.0.0, that is :)